pkgs/development/libraries/imgui/default.nix

```diff
@@ -1,8 +1,77 @@
-{ stdenv, lib, fetchFromGitHub }:
+{
+  stdenv,
+  lib,
+  applyPatches,
+  callPackage,
+  cmake,
+  fetchFromGitHub,
+  fetchpatch,
+  glfw,
+  libGL,
+  darwin,
+  vcpkg,
+  SDL2,
+
+  # The intent is to mirror vcpkg's flags[^1],
+  # but we limit our attention to Linux and glfw3 for now
+  # [^1]: https://github.com/microsoft/vcpkg/blob/095ee06e7f60dceef7d713e3f8b1c2eb10d650d7/ports/imgui/CMakeLists.txt#L33-L108
+  withGLLoader ? !libGL.meta.unsupported,
+  withFeatures ?
```

Comment:
Not allowing to overwrite a single decision but only everything is no that great

Reply:
The passthru tests include an example which uses e.g. `unique (prev.withFeatures ++ [ ... ])` but that is clearly much less flexible than individual attributes.

One obvious alternative is to make the lambda take `IMGUI_BUILD_GLFW_BINDING, ...` as arguments, directly mirroring vcpkg's options. The issue with that is I'd have to be selective: I wouldn't want to declare `IMGUI_BUILD_ALLEGRO5_BINDING` or `IMGUI_BUILD_DX12_BINDING` because I don't actually plan to be implementing support for that (someone else might, I won't).

I could really use some advise or suggestions on this one

Reply:
Ditched lists and passthru in favour of boilerplate attributes